src/plugins/data/server/plugin.ts

```diff
@@ -72,7 +73,7 @@ export class DataServerPlugin
   private readonly scriptsService: ScriptsService;
   private readonly kqlTelemetryService: KqlTelemetryService;
   private readonly autocompleteService: AutocompleteService;
-  private readonly indexPatterns = new IndexPatternsService();
+  private readonly indexPatterns = new IndexPatternsServiceProvider();
```

Comment:
Its not clear to me why this was renamed

Reply:
Because otherwise there are two things called "index patterns service" whereas this renamed one is not actually an index pattern service but rather something that constructs one.